clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java

```diff
@@ -76,4 +80,59 @@ public void testClientMode() throws Exception {
         assertTrue(engine.getUseClientMode());
     }
 
+    @Test
+    public void testKeyStoreTrustStoreValidation() throws Exception {
```

Comment:
Should we have a basic sanity test case for reconfiguration itself? I looked for one and couldn't find any.

Reply:
Added in `SslTransportLayerTest`.